core/valor_core/classification.py

```diff
@@ -0,0 +1,1343 @@
+    unique_labels : list
+        List of unique labels used for the classification.
+    label_map : Optional[Dict[schemas.Label, schemas.Label]], default=None
+        Optional dictionary mapping ground truth labels to prediction labels.
+    metrics_to_return : Optional[List[enums.MetricType]], default=None
+        List of metric types to return. If None, default metrics are used.
+    pr_curve_max_examples : int, default=1
+        Maximum number of examples to use for Precision-Recall curve calculations.
+
+    Returns
+    -------
+    Tuple[List[dict], List[dict]]
+        A tuple where:
+        - The first element is a list of dictionaries representing confusion matrices.
+        - The second element is a list of dictionaries representing the requested classification metrics.
+    """
+    grouper_mappings = _create_classification_grouper_mappings(
+        label_map=label_map,
+        labels=unique_labels,
+    )
+
+    confusion_matrices, metrics_to_output = [], []
+
+    _add_columns_to_groundtruth_and_prediction_table(
+        df=groundtruth_df, grouper_mappings=grouper_mappings
+    )
+    _add_columns_to_groundtruth_and_prediction_table(
+        df=prediction_df, grouper_mappings=grouper_mappings
+    )
+
+    merged_groundtruths_and_predictions_df = _get_joint_df(
+        prediction_df=prediction_df, groundtruth_df=groundtruth_df
+    )
+
+    cm_counts_df, confusion_matrices = _calculate_confusion_matrix_df(
+        merged_groundtruths_and_predictions_df=merged_groundtruths_and_predictions_df
+    )
+
+    metrics_per_grouper_key_and_grouper_value_df = (
+        _calculate_metrics_at_grouper_value_level(cm_counts_df=cm_counts_df)
+    )
+
+    metrics_to_output += _calculate_precision_recall_f1_metrics(
+        metrics_per_grouper_key_and_grouper_value_df=metrics_per_grouper_key_and_grouper_value_df
+    )
+
+    metrics_to_output += _calculate_accuracy_metrics(cm_counts_df=cm_counts_df)
+
+    metrics_to_output += _calculate_rocauc(
+        prediction_df=prediction_df, groundtruth_df=groundtruth_df
+    )
+
+    # handle type error
+    assert metrics_to_return
+
+    metrics_to_output += _calculate_pr_curves(
+        prediction_df=prediction_df,
+        groundtruth_df=groundtruth_df,
+        metrics_to_return=metrics_to_return,
+        pr_curve_max_examples=pr_curve_max_examples,
+    )
+
+    # convert objects to dictionaries and only return what was asked for
+    metrics_to_output = [
+        m.to_dict()
+        for m in metrics_to_output
+        if m.to_dict()["type"] in metrics_to_return
+    ]
+    confusion_matrices = [cm.to_dict() for cm in confusion_matrices]
+
+    return confusion_matrices, metrics_to_output
+
+
+def evaluate_classification(
+    groundtruths: Union[pd.DataFrame, List[schemas.GroundTruth]],
+    predictions: Union[pd.DataFrame, List[schemas.Prediction]],
+    label_map: Optional[Dict[schemas.Label, schemas.Label]] = None,
```

Comment:
Expanding on my comment above, i feel that label mapping should occur in the data aggregation layer and not within the computation itself. This will reduce a lot of complexity in the computation and allow for tighter testing coverage.

Reply:
see above